File: StrykerDG.FarmForge.Client/farmforge_client/lib/models/inventory/product_category.dart
class ProductCategory {
  int productCategoryId;
  String name;
  String label;
  String description;

  ProductCategory({
    this.productCategoryId,
    this.name,
    this.label,
    this.description
  });

  factory ProductCategory.fromMap(Map<String, dynamic> data) {
    return ProductCategory(
      productCategoryId: data['ProductCategoryId'],
      name: data['Name'],
      label: data['Label'],
      description: data['Description']
    );
  }

  Map<String, dynamic> toMap() {
    return {
      'ProductCategoryId': this.productCategoryId,
      'Name': this.name,
      'Label': this.label,
      'Description': this.description
    };
  }
}

File: StykerDG.FarmForge.DataModel/Models/ProductCategory.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Text;

namespace StrykerDG.FarmForge.DataModel.Models
{
    [Table("ProductCategory")]
    public class ProductCategory : BaseModel
    {
        [Key]
        public int ProductCategoryId { get; set; }
        public string Name { get; set; }
        public string Label { get; set; }
        public string Description { get; set; }
    }
}
